### PR TITLE
[Fix] Fix mmdet ort static shape bug.

### PR DESCRIPTION
**Pull request by hanrui1sensetime**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix bugs mentioned in #679 

## Modification

base_dense_head.py

## BC-breaking (Optional)

None.

## Use cases (Optional)

None.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/codebase/mmdet/models/dense_heads/base_dense_head.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def base_dense_head__get_bbox(ctx,
         ]
         mlvl_score_factors = []
     assert img_metas is not None
-    img_shape = img_metas[0]['img_shape']
+    img_shape = torch.Tensor(img_metas[0]['img_shape'])
```
Comment (grimoire) [inline mmdeploy/codebase/mmdet/models/dense_heads/base_dense_head.py:83]: use `torch.tensor` and don't forgot to set the device.

File: mmdeploy/codebase/mmdet/models/dense_heads/base_dense_head.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def base_dense_head__get_bbox(ctx,
         ]
         mlvl_score_factors = []
     assert img_metas is not None
-    img_shape = img_metas[0]['img_shape']
+    img_shape = torch.tensor(
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmdet/models/dense_heads/base_dense_head.py:83]: since we have rewriting for `distance2bbox`, maybe we don't need to change here.

**Comment by RunningLeon** [comment]
@hanrui1sensetime Could fix UT `test_distance2bbox` which failed in CI

File: mmdeploy/codebase/mmdet/core/bbox/distance_point_bbox_coder.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import mmdet.core.bbox.transforms
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmdet.core.bbox.coder.DistancePointBBoxCoder.decode',
+    backend='default')
+def decode__default(ctx, self, points, pred_bboxes, max_shape=None):
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmdet/core/bbox/distance_point_bbox_coder.py:10]: This naming style is not aligned with others. Besides, should add ut for this rewriting.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by grimoire — APPROVED** [review]

**Pull request merged**